File: ABC122/C.hs
module Main where

import Control.Monad (replicateM_)

count :: String -> [Int]
count s = count' 0 $ zip s $ tail s
  where
    count' n [] = [n]
    count' n (('A','C'):rest) = n:count' (n+1) rest
    count' n (_:rest) = n:count' n rest

main :: IO ()
main = do
  [_, q] <- map read . words <$> getLine -- skip n
  s <- getLine
  let acc = count s
  replicateM_ q $ do
    [l, r] <- map read . words <$> getLine
    print $ acc!!(r-1) - acc!!(l-1)

File: ABC122/C.cpp
#include <iostream>
#include <string>
#include <vector>

int main() {
  int N, Q;
  std::cin >> N >> Q;

  std::string S;
  std::cin >> S;

  std::vector<int> acc(N);

  for (int i=1; i<N; i++) {
    acc[i] = acc[i-1];
    if (S[i-1] == 'A' && S[i] == 'C') acc[i]+=1;
  }

  for (int i=0; i<Q; i++) {
    int l, r;
    std::cin >> l >> r;
    std::cout << acc[r-1] - acc[l-1] << '\n';
  }

  return 0;
}
